src/client.ts

```diff
@@ -1513,6 +1513,21 @@ export class StreamChat<
     }
     // support channel("messaging", {options})
     if (typeof channelIDOrCustom === 'object') {
```

Comment:
**Existing logic:**
We maintain a list of active channels on client - `client.activeChannel`. 

So every-time, user requests a channel using `client.channel('messaging', 'random_channel_id')`, we first [check](https://github.com/GetStream/stream-chat-js/blob/5f1669925f3ff22e4948f28bd0e81aa218aafe6a/src/client.ts#L1550) if the `cid` already exist in `client.activeChannels`
--> if yes, then return the channel available on client.activeChannels
--> if not, then create new instance of Channel class and return that.

This way we ensure that user doesn't end up creating multiple copies of same channel. And this is important, because when client receives ws event, it delegates it to all channels in `activeChannel` map. So if your channel isn't part of `activeChannels`, it won't receive any of the events.

**Issue**

This logic only worked for id based channels. If user requests a member based channel (e.g., `client.channel('messaging', { members: ['vishal', 'amin'] })`), then we skipped that check, because the logic for generating id from members list is on backend side, and its not a good idea to expose it (checked with @thesyncim ) on frontend. So we couldn't really check if the channel already exists in `client.activeChannels`

**Fix**

We are adding this check for member based channels. We assert that a particular channel in `activeChannels` is same as requested channel by user:

1. If `cid` of channel starts with `${channelType}:!members-`
2. channel has same members as the channel which is requested.